### DISCUSSION TITLE
NoneType' object is not subscriptable' - Training Failed

### DISCUSSION THREAD

**Discussion by Anashel-RPG** [d8ahazard/sd_dreambooth_extension#799] [Q&A]
Greetings!

First incredbile extension. If there is any way I can support your financially for all the work, please let me know. I am running the following JSON format for my concept:

{
"max_steps": -1,
"instance_data_dir": "PATH_TO_IMAGE",
"class_data_dir": "PATH_TO_IMAGE",
"num_class_images": 200,
"instance_prompt": "[filewords]",
"class_prompt": "creatures",
"class_negative_prompt": "woman, human, men, person",
"save_sample_prompt": "a digital painting of a mighty orc warrior creature"
}

This exact JSON (with about 10 concepts) worked perfectly for a while but sadly stopped working when I did the Git Pull today. The rest of my config is as follow (Running on A10 AWS instance)
---
Steps per image: 150
Pause: 0
Time to pause: 0
Save model Frequency: 15
Save previews: 5
Batch Size: 4
Gradient Acc Steps: 4
Class Batch: 4
Set Gradients: True
Gradient Checkpoint: True
Learning Rate: Constat with Warmup
Learning Rate: 0.0000014
Learning Rate Warmup: 500
Resolution: 512
(No other changes for these until the row Use Copncepts)
Use Concepts: True
Concepts List (The JSON I just posted above)
Use EMA: False
Use 8bit: True
Mixed Precision: fp16
Memory: xformers
(All other setting left as is)

For the Saving Tabs:
Half Model: True
Save in subdirectory: True
Generate .CKPT when saving training: True
Generate .CKPT when training over: True

**Comment by Anashel-RPG** [comment]
Worth mentioning that Tab "Generate" trigger the same error when clicking: generate sample images. "Exception generating sample(s): 'NoneType' object has no attribute 'device'"

**Comment by Anashel-RPG** [comment]
Setting the value Mixed Precision to "no" skip everything and end with "Training finished. Total lifetime steps: 0". When set to fp16, the error is ''NoneType' object is not subscriptable'.

**Comment by ZhenyaPav** [comment]
Did you manage to fix this issue?

**Comment by Prometheus3D** [comment]
I'm on Windows and had to revert back to commit **534baa08**, which was prior to some "MacOS fixes" (**6f872348**).
Alternatively, you can try to encode all your training text files to UTF-8.

It seems like the code change afterward was less accepting of ASCII encoding as suggested here [https://github.com/d8ahazard/sd_dreambooth_extension/issues/893](url)


Someone please correct or elaborate on something I may have gotten wrong.
